src/pfe/portal/modules/User.js

```diff
@@ -746,19 +750,25 @@ module.exports = class User {
       return registrySecretList;
     }
 
-    const msg = "Failed to patch the Service Account";
+    const msg = "Reverting changes to the Docker Config";
     log.error(msg);
-    throw new RegistrySecretsError("SERVICE_ACCOUNT_PATCH_FAILED", "for address " + address);
+    
+    // Since patching the Service Account failed, we need to revert the update to the Docker Config and patch the Service Account again
+    delete jsonObj.auths[address];
+    await fs.writeJson(this.dockerConfigFile, jsonObj);
+    await this.updateServiceAccountWithDockerRegisrySecret();
+
+    throw new RegistrySecretsError(serviceAccountPatchData.error, "for address " + address);
   }
 
   /**
-   * Function to create the Kubernetes secret and patch the Service Account
+   * Function to create the Kubernetes Secret and patch the Service Account with the created Secret
    */
   async updateServiceAccountWithDockerRegisrySecret() {
```

Comment:
This function is a little odd now, it either returns an object that contains `true` or one that contains `false` and an error message. Would it be easier to return an error message on failure and `undefined`/`null` on success?
You could also throw the error but I can see why that might be tricky with lots of different callers.

It's a bit odd to return an error or nothing in Node.js but C does it all the time with non-zero return codes. ;-)

Reply:
I'll update it to be either undefined or ERR_CODE